src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2622,6 +2557,70 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
             return true;
         }
 
+        private static Stream GetAuxPeStream(DiagnosticBag metadataDiagnostics, EmitStreamProvider metadataPEStreamProvider)
+        {
+            if (metadataDiagnostics.HasAnyErrors())
+            {
+                return null;
+            }
+
+            var auxStream = metadataPEStreamProvider.GetOrCreateStream(metadataDiagnostics);
+            Debug.Assert(auxStream != null || metadataDiagnostics.HasAnyErrors());
+            return auxStream;
+        }
+
+        /// <summary>
+        /// Returns a tuple of streams where
+        /// * `peStream` is a stream which will carry the output PE bits
+        /// * `signingStream` is the stream which will be signed by the legacy strong name signer, or null if we aren't using the legacy signer
+        /// * `selectedStream` is an alias of either peStream or signingStream, and is the stream that will be written to by the emitter.
+        /// </summary>
+        private (Stream peStream, Stream signingStream, Stream selectedStream) GetPeStream(DiagnosticBag metadataDiagnostics, EmitStreamProvider peStreamProvider, bool metadataOnly)
```

Comment:
I would call it something other than `GetPeStream` since it's not just returning a `peStream`

Reply:
`GetPeStreamAndMaybeTheSigningStreamToo`?

Reply:
`GetPeStreams`?